test/Makefile

```diff
@@ -38,4 +45,16 @@ test_cheap:
 	cd sharness && make
 	cd 3nodetest && make
 
-.PHONY: all clean
+test_race:
+	cd sharness && make GOFLAGS=-race TEST_EXPENSIVE=1
+	cd 3nodetest && make GOFLAGS=-race
+	cd dependencies && make GOFLAGS=-race
+
+IPFS-BUILD-OPTIONS: FORCE
+	@VARS='$(GOFLAGS)'; \
+	    if test x"$$VARS" != x"`cat $@ 2>/dev/null`" ; then \
+		echo "*** new Go flags ***"; \
+		echo "$$VARS" >$@; \
+            fi
```

Comment:
this seems very magic-y to me. wonder if maybe the right thing to do is to have both:

```
bin/program
bin/program-race
```
that way we could even have concurrent tests going